app.js

```diff
@@ -0,0 +1,157 @@
+//  in this file create an express application - use the middle-ware built into express
+//  to serve up static files from the public directory (index.html and client.js - you 
+//  can also serve up css files from the public directory this way if you want)
+//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks:name'
+
+/*
+*  For Module 4b
+*  1.  A GET route for '/trucks' that sends to the response the list of all food-truck objects in 
+*		JSON format.
+*  2.  A GET route for '/trucks/:name' that sends to the response the matching food-truck object in 
+*		JSON format.
+*  3.  A GET route for '/food-types' that sends to the response the unique list of food types 
+*		associated with the food trucks in the foodTrucks list. And in the same app.js file, add 
+*		the following three new routes:
+*  4.  A POST route for '/trucks' that posts a new food-truck object consisting of (if they exist) 
+*		the following values. After creating a new food truck object, add it to the foodTrucks 
+*		array, then send it to the response: name food type schedule of days payment types accepted 
+*		website url Facebook page url Twitter url
+*  5.  A DELETE route for '/trucks/:name' that sends a delete request that will delete the 
+*		corresponding food-truck object from the foodTrucks array.
+*/
+var express = require('express');  //require the express module which returns an object/function
+var app = express(); //creates and express application instance
+
+var bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(bodyParser.json());
+
+
+
+var serveStatic = express.static('public');
+
+app.use(serveStatic);
+
+var trucks = require('./trucks');
+var truckObject = trucks();
+
+app.get('/trucks',function(request,response){
+	var truckList = truckObject.getTrucks();
+	response.send(truckList);
+});
+
+
+
+app.get('/trucks/:name',function(request,response){
+
+  //grab the name from the url ex)http://localhost:3000/trucks/TedsTruck gets foodTruck='TedsTruck'
+  //name, food-type, payment methods accepted, description, website, and schedule.
+  var foodTruckName = request.params.name;
+  var foodTruck = truckObject.getTruck(foodTruckName);
+  var truckHTML = '';
+  var truckName = foodTruck.name; //string
+  
+  var foodType = foodTruck.type; //array
+  var foodString = '';
+  if(foodType){
+	// build the foodString - taking into account if blank, one entry, or many
+	if(foodType.length===0){
+		foodString = 'no entry found';
+	}else if(foodType.length>0){
+		foodString = foodType[0];
+		if(foodType.length>1){
+			for(i=1;i<foodType.length;i++){
```

Comment:
You can also do:

`foodString = foodType.join(',');`

here to get the same result.